src/main.rs

```diff
@@ -390,17 +390,21 @@ fn run() -> Result<(), failure::Error> {
     } else if let Some(matches) = matches.subcommand_matches("generate") {
         let name = matches.value_of("name").unwrap_or("worker");
         let site = matches.is_present("site");
+        let template_value = matches.value_of("template");
         let mut target_type = None;
 
         let template = if site {
+            if template_value.is_some() {
+                failure::bail!("You cannot specify a template when generating a Workers Site. If you want to generate site boilerplate, run wrangler generate --site")
```

Comment:
This text is a little confusing to me- I might say:

"You cannot specify a template when generating a Workers Site. If you'd like to use the default site boilerplate, run `wrangler generate --site`. If you'd like to use another site boilerplate, omit the `--site` when running `wrangler generate`."